web/src/lib/components/asset-viewer/photo-editor.svelte

```diff
@@ -0,0 +1,629 @@
+    let maxHeight: string = String(window.innerHeight);
+    console.log(maxWidth, maxHeight);
+    imageEditor = new ImageEditor(editorElement, {});
+    console.log(imageEditor.getCanvasSize() as any);
+    try {
+      await loadAssetData();
+    } catch (error) {
+      // Throw error
+      handleError(error, 'Failed to load asset data');
+    }
+    const result = await imageEditor.loadImageFromURL(assetData, 'test');
+    console.log(result);
+  });
+
+  const loadAssetData = async () => {
+    try {
+      const { data } = await api.assetApi.serveFile(
+        { id: asset.id, isThumb: false, isWeb: true, key: publicSharedKey },
+        {
+          responseType: 'blob',
+        },
+      );
+
+      if (!(data instanceof Blob)) {
+        return;
+      }
+
+      assetData = URL.createObjectURL(data);
+      return assetData;
+    } catch {
+      // Do nothing
+      console.log('Failed to load asset data');
+    }
+  };
+
+  const navigateEditTab = async (button: 'autofix' | 'crop' | 'adjust' | 'filter') => {
+    activeButton = button;
+    switch (activeButton) {
+      case 'autofix':
+        break;
+      case 'crop':
+        await setAspectRatio('free');
+        initAngleSlider();
+        break;
+      case 'adjust':
+        break;
+      case 'filter':
+      default:
+        break;
+    }
+  };
+
+  const setAspectRatio = async (ratio: aspectRatio) => {
+    aspectRatio = ratio;
+    // Switch for all aspect ratios
+    await imageEditor.startDrawingMode('CROPPER');
+    switch (aspectRatio) {
+      case 'free':
+        imageEditor.setCropzoneRect();
+        break;
+      case 'square':
+        imageEditor.setCropzoneRect(1 / 1);
+        break;
+      case 'original':
+        // const canvasSize = imageEditor.getCanvasSize();
+        // imageEditor.setCropzoneRect(canvasSize.width / canvasSize.height);
+        imageEditor.setCropzoneRect();
+        break;
+      case '16_9':
+        imageEditor.setCropzoneRect(16 / 9);
+        break;
+      case '9_16':
+        imageEditor.setCropzoneRect(9 / 16);
+        break;
+      case '5_4':
+        imageEditor.setCropzoneRect(5 / 4);
+        break;
+      case '4_5':
+        imageEditor.setCropzoneRect(4 / 5);
+        break;
+      case '4_3':
+        imageEditor.setCropzoneRect(4 / 3);
+        break;
+      case '3_4':
+        imageEditor.setCropzoneRect(3 / 4);
+        break;
+      case '3_2':
+        imageEditor.setCropzoneRect(3 / 2);
+        break;
+      case '2_3':
+        imageEditor.setCropzoneRect(2 / 3);
+        break;
+      default:
+        imageEditor.setCropzoneRect();
+        break;
+    }
+  };
+
+  //function for dragging the angle selection slider
+  const initAngleSlider = async () => {
+    console.log('initAngleSlider');
+    let pos1 = 0,
+      pos2 = 0,
+      pos3 = 0,
+      pos4 = 0;
+
+    const closeDragElement = () => {
+      // stop moving when mouse button is released:
+      document.onmouseup = null;
```

Comment:
Probably not great to delete event listeners globally. We should instead use `addEventListener` and `removeEventListener`.

Expanding further, I wonder if it actually makes sense to bind this to a `<svelte:window />` element? A lot of the race conditions and management can then be deferred to svelte. Such:

```svelte
<div bind:this={angleSliderHandle} on:mousedown={handleMouseDown} />
```

```svelte
{#if mouseDown
<svelte:window on:mouseup={handleMouseUp} on:mousemove={handleMouseMove} />
}
```